pytorch_lightning/utilities/data.py

```diff
@@ -336,14 +353,29 @@ def wrapper(obj: DataLoader, *args: Any, **kwargs: Any) -> None:
             if param.name != "self" and param.kind not in (param.VAR_POSITIONAL, param.VAR_KEYWORD)
         ]
 
-        cls = type(obj)
-        for arg_name, arg_value in chain(zip(param_names, args), kwargs.items()):
-            if hasattr(cls, arg_name) and getattr(cls, arg_name).fset is None:
-                # the class defines a read-only (no setter) property of this name. it's likely that the implementation
-                # will set `self._arg_name = arg_value` in `__init__` which is the attribute returned by the `arg_name`
-                # property so we are fine skipping in that case
-                continue
-            setattr(obj, arg_name, arg_value)
+        if not hasattr(obj, "_set_arg_names"):
+            set_arg_names = set()
+            for arg_name, arg_value in chain(zip(param_names, args), kwargs.items()):
+                if not hasattr(obj, "__" + arg_name):
```

Comment:
Could it happen that `hasattr(obj, "_set_arg_names") is True` but `hasattr(obj, "__" + arg_name) is False`?

As in, is this second if redundant?

Reply:
You're right, it was redundant, I removed it now